GitExtensions/Program.cs

```diff
@@ -54,6 +54,11 @@ private static void Main()
                 }
             }
 
+            // This is done here so these values can be used in the GitGui project but this project is the authority of the values.
+            UserEnvironmentInformation.Sha = ThisAssembly.Git.Sha;
+            UserEnvironmentInformation.Dirty = ThisAssembly.Git.IsDirty;
+            UserEnvironmentInformation.LockValues();
```

Comment:
I'd change the API to something like:

```
UserEnvironmentInformation.Initialise(ThisAssembly.Git.Sha, ThisAssembly.Git.IsDirty);
```

```
        private static bool _alreadySet;
        private static bool _dirty;
        private static string _sha;

    public static void Initialise(string sha, bool isDirty)
    {
        if (_alreadySet) { return;  // or throw? }
        _alreadySet = true;
        _sha = sha;
        _dirty = isDirty;
    }

    public static string GetInformation()
    {
        if (!_alreadySet) { throw new InvalidOperationException($"{nameof(Init)} must be called first"); }
```

This way the public API surface remains minimal.

Reply:
Done.